aws-tf/terraform-provider-aws-parallelcluster
# Makefile updates

### Description of changes
This PR adds common targets to the makefile
  - add build, uninstall, and clean targets
  - add generated test targets
  - allow setting version from env, ie `VERSION=3.9.0-2 make install`
  - Remove mention of acceptance tests
  - add underscores to test targets

### Tests
* Run `VERSION=3.9.0-2 make install` and verify binary is installed with the expected version and location
* Run `make uninstall` and verify provider binaries are removed
* Run `make clean` and verify the build folder is cleaned

### Checklist
By submitting the PR you confirm that you are going to do what follows before merging your code.
1. I have read [CONTRIBUTING](../CONTRIBUTING.md).
2. I have made the PR point to **the right branch**.
3. I have added the right labels to the PR.
4. I have checked that all commits' messages are clear, describing what and why vs how.
5. I have successfully executed lint and tests to prove the quality and correctness of the change.
6. I have added tests to validate the proposed change.
7. I have added the documentation to describe my changes (if appropriate).
8. I have added the necessary entries to the changelog (if appropriate).
9. Any dependent changes have been merged and published in downstream modules.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
